src/pyTRLCConverter/markdown_converter.py

```diff
@@ -686,7 +693,11 @@ def markdown_create_diagram_link(diagram_file_name: str, diagram_caption: str, e
         if escape is True:
             diagram_caption_raw = MarkdownConverter.markdown_escape(diagram_caption)
 
-        return f"![{diagram_caption_raw}](./{diagram_file_name})\n"
+        # Filenames are relative to the Markdown file.
+        if not diagram_file_name.startswith("./"):
```

Comment:
Relative path  is not a requirement for markdown. I would also check also if the path is relative and not absolute like  "/tmp/x.png". But I agree that absolute is likely not usefull, but its possible.